tests/integration/test_dot_venv.py

```diff
@@ -41,15 +41,15 @@ def test_reuse_previous_venv(PipenvInstance, pypi):
         assert c.return_code == 0
         assert normalize_drive(p.path) in p.pipenv('--venv').out
 
-            
+
 def test_venv_file_exists(PipenvInstance, pypi):
     """Tests virtualenv creation & package installation when a .venv file exists
     at the project root.
     """
     with PipenvInstance(pypi=pypi, chdir=True) as p:
         file_path = os.path.join(p.path, '.venv')
         with open(file_path, 'w') as f:
-            f.write('test-project')
+            f.write('')
```

Comment:
The merge should have kept `test_venv_file_exists' over `test_venv_file_exists` (or kept both). We do want to write something to the file so we can test if the written name is used to find/create a venv of the same name in workon home.

Reply:
hmmmmm I think I prioritized your test over the one he wrote, but that wasn't really intentional, I was just trying to triage and get everything moving so I can get a release out soon(tm). There is a test right below that one though and it does write a name of a directory to the file which is why I undid my inclusion of it here

The merge removed this project name in favor of creating a temporary directory in the test below it and using that directory's location as the path in the `,venv` file.

Reply:
The difference between the two tests was one wrote a path to the file, the other just a name. If you think that the former adequately covers the latter case too then all is fine as is.

Reply:
I think those should be merged into one with pytest fixtures. The fixture would be something like

```python
@pytest.mark.fixtures('content, venv_location', [
    ('just-a-name', Path(WORKON_HOME, 'just-a-name')),
    ('/an/absolute/path', Path('/an/absolute/path')),
    ('a/relative/path', Path(project_root, 'a/relative/path')),
])
```

(I added a third test case. If a relative path is specified, the venv should be created relative to the project’s root.)

Reply:
Good call, but does it really need to be a fixture if we'll only have one test that uses it? Could just use `@pytest.mark.parametrize`. 

I'm not sure how `WORKON_HOME` and `project_root` be passed. My first thought was fixtures, but pytest currently doesn't support the use of fixtures in `@pytest.mark.parametrize`. Would it work if it was left a fixture, and a fixture was used for the fixture's params?

Reply:
Ah yes, excellent call, I was totally thinking about `parametrize`, not fixtures!

Reply:
I don’t think we need to refactor tests to merge this PR but the point about a path vs a name is valid. You can’t parametrize a temporary directory as an input though so that would still need a separate test

Reply:
Yeah, I played around with this idea of parametrisation but couldn't come up with any elegant solution - it'd create a bigger mess than it is intended to prevent.

Reply:
Do you mind the messy commit history? I could just commit over everything to make the test write a name again, or I could just rewrite history and re-do the merge.

Reply:
I usually squash the commits a bit before merging, so make it as messy as you need to :p